exchanges/binanceus/binanceus_websocket.go

```diff
@@ -42,93 +41,93 @@ var (
 )
 
 // WsConnect initiates a websocket connection
-func (b *Binanceus) WsConnect() error {
-	if !b.Websocket.IsEnabled() || !b.IsEnabled() {
+func (bi *Binanceus) WsConnect() error {
+	if !bi.Websocket.IsEnabled() || !bi.IsEnabled() {
 		return errors.New(stream.WebsocketNotEnabled)
 	}
 	var dialer websocket.Dialer
-	dialer.HandshakeTimeout = b.Config.HTTPTimeout
+	dialer.HandshakeTimeout = bi.Config.HTTPTimeout
 	dialer.Proxy = http.ProxyFromEnvironment
 	var err error
-	if b.Websocket.CanUseAuthenticatedEndpoints() {
-		listenKey, err = b.GetWsAuthStreamKey(context.TODO())
+	if bi.Websocket.CanUseAuthenticatedEndpoints() {
+		listenKey, err = bi.GetWsAuthStreamKey(context.TODO())
 		if err != nil {
-			b.Websocket.SetCanUseAuthenticatedEndpoints(false)
+			bi.Websocket.SetCanUseAuthenticatedEndpoints(false)
 			log.Errorf(log.ExchangeSys,
 				"%v unable to connect to authenticated Websocket. Error: %s",
-				b.Name,
+				bi.Name,
 				err)
 		} else {
 			// cleans on failed connection
-			clean := strings.Split(b.Websocket.GetWebsocketURL(), "?streams=")
+			clean := strings.Split(bi.Websocket.GetWebsocketURL(), "?streams=")
 			authPayload := clean[0] + "?streams=" + listenKey
-			err = b.Websocket.SetWebsocketURL(authPayload, false, false)
+			err = bi.Websocket.SetWebsocketURL(authPayload, false, false)
 			if err != nil {
 				return err
 			}
 		}
 	}
-	err = b.Websocket.Conn.Dial(&dialer, http.Header{})
+	err = bi.Websocket.Conn.Dial(&dialer, http.Header{})
 	if err != nil {
 		return fmt.Errorf("%v - Unable to connect to Websocket. Error: %s",
-			b.Name,
+			bi.Name,
 			err)
 	}
 
-	if b.Websocket.CanUseAuthenticatedEndpoints() {
-		go b.KeepAuthKeyAlive()
+	if bi.Websocket.CanUseAuthenticatedEndpoints() {
+		go bi.KeepAuthKeyAlive()
```

Comment:
There is waitgroup Add and Close method call inside the method

Reply:
Yeah you are correct, It's a convention to call the `wg.Add(1)` before a go routine call because there could be a delay in the go schedular and it might cause a routine leak. We had a lot of instances throughout this codebase.

Reply:
bump